src/averageAges.js

```diff
@@ -55,7 +65,23 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = (arguments.length === 2)
+    ? people.filter(child => child.sex === 'm'
+      && people.some(mother => mother.name === child.mother))
+    : people.filter(child => people.some(
+      mother => mother.name === child.mother));
+
+  children.every(child => {
+    const mother = people.find(person => person.name === child.mother);
+
+    child.motherSAgeAtBorn = child.born - mother.born;
+
+    return true;
+  });
```

Comment:
Use different array method.
Try not to mutate anything while using non-mutating array methods, like map, filter, some or every.